napari/components/dims.py

```diff
@@ -159,17 +165,49 @@ def nsteps(self) -> Tuple[int, ...]:
             for min_val, max_val, step_size in self.range
         )
 
+    @nsteps.setter
+    def nsteps(self, value):
+        self.set_range(
+            range(self.ndim),
+            tuple(
+                (mn, mx, (mx - mn) / val)
+                for (mn, mx), val in zip(self.range, value)
+            ),
+        )
+
     @property
-    def point(self) -> Tuple[int, ...]:
-        """Tuple of float: Value of each dimension."""
-        # The point value is computed from the range and current_step
-        point = tuple(
-            min_val + step_size * value
-            for (min_val, max_val, step_size), value in zip(
-                self.range, self.current_step
-            )
+    def thickness(self) -> Tuple[float]:
+        return tuple(high - low for low, high in self.span)
+
+    @thickness.setter
+    def thickness(self, value):
+        self.set_thickness(range(self.ndim), value)
+
+    @property
+    def point(self) -> Tuple[float, ...]:
+        return tuple((low + high) / 2 for low, high in self.span)
+
+    @point.setter
+    def point(self, value):
+        self.set_point(range(self.ndim), value)
+
+    @property
+    def current_step(self) -> Tuple[int, ...]:
+        return tuple(
+            int(round((point - min_val) / step))
+            for point, (min_val, _, step) in zip(self.point, self.range)
+        )
+
+    @current_step.setter
+    def current_step(self, value: Tuple[int, ...]):
+        warnings.warn(
+            trans._(
+                'Dims.current_step is deprecated. Use Dims.set_point_step instead.'
+            ),
+            DeprecationWarning,
+            stacklevel=2,
         )
-        return point
+        self.set_point_step(range(self.ndim), value)
```

Comment:
I agree with @andy-sweet that this API change isn't great. The main reason why we are using this pattern is because we used to lack evented lists, but now that we have that object, maybe we should make all these attributes evented lists, and then users can just do `viewer.dims.current_step[axis] = step`?